source/DataAccess/Oracle/OracleDDLHelper.ts

```diff
@@ -3,24 +3,26 @@ import { injectable } from 'inversify';
 
 @injectable()
 export default class OracleDDLHelper implements IDDLHelper {
-    public getDDLQuery(tableNames: string[]): string {
-        const tableFilter = tableNames.map((tbl) => `'${tbl}'`).join(',');
-
+    public getDDLQuery(validTableNames: string[]): string {
         // Having the ability to use DBMS_METADATA.SET_TRANSFORM_PARAM would help here to remove things like
         // disabling the scripting of the 'user' in create statements but this system function does not seem
         // to always be available despite 'GET_DDL' working
         // NOTE: Scripting dependencies (FKs etc) via simple commands such as GET_DDL does not seem to be
         // possible and would require some extra overhead to implement
-        const ddlQuery =
-            'SELECT 1 as "priority", u.table_name, DBMS_METADATA.GET_DDL(\'TABLE\',u.table_name) as ddl ' +
-            'FROM USER_TABLES u ' +
-            `WHERE u.table_name in (${tableFilter})` +
-            'union all ' +
-            'SELECT 2 as "priority", u.table_name, DBMS_METADATA.GET_DDL(\'INDEX\',u.index_name) as ddl ' +
-            'FROM USER_INDEXES u ' +
-            `WHERE u.table_name in (${tableFilter})`  +
-            'ORDER BY table_name, "priority"';
-
-        return ddlQuery;
+        return validTableNames
```

Comment:
Knowing what tables we were able to fetch is essential to this query at least giving us back the table DDL. The change to call the function based on this list of tables versus querying the user_tables table is an effort to reduce the footprint of permissions we request of a client. I am not sure we can always rely on a client to provide access to these underlying tables whereas I think the minimal requirement to have access to DBMS_METADATA is pretty much non-negotiable.

Reply:
Ah; Makes sense. Did we get this feedback from the customer?
@lsanders might also have more feedback on the most appropriate way to get DDL from Oracle, but this does seem like it would be less intrusive than user_tables. Out of curiousity, it might be worth asking the customer if they're willing to grant read access to user_tables for now to help speed things up.